src/averageAges.js

```diff
@@ -55,7 +62,15 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const onlyChid = !onlyWithSon
+    ? people.filter(el => people.find(child => el.mother === child.name))
+    : people.filter(el => (people.find(child => el.mother === child.name
+     && el.sex === 'm')));
```

Comment:
```suggestion
    ? people.filter(child => people.find(mother => child.mother === mother.name))
    : people.filter(child => (people.find(mother =>child.mother === mother.name
     && child.sex === 'm')));
```
I offer to change variables names like that to make your code more understandable